### PR TITLE
[microTVM] Bump Arduino SDK versions in Vagrant VM

### PR DESCRIPTION
**Pull request by guberti**
In the past two weeks, a new version of the Arduino SDK (`v0.1.9`) and a new version of the Spresense SDK (`v2.3.0`) have been released. These fix two major bugs that caused problems for TVM (thanks to @mdw-octoml for originally reporting these over six months ago), which lets us remove two ugly hacks.

I'm currently testing these changes on Wolfi to ensure they don't break anything - once all looks good, I'll remove the "draft" status from this pull request.

### CODE REVIEW

File: apps/microtvm/reference-vm/arduino/base-box/base_box_provision.sh [hunk 1]
```diff
@@ -45,7 +45,7 @@ ADAFRUIT_BOARDS_URL="https://adafruit.github.io/arduino-board-index/package_adaf
 ESP32_BOARDS_URL="https://raw.githubusercontent.com/espressif/arduino-esp32/gh-pages/package_esp32_dev_index.json"
```
Comment (mehrdadh) [inline apps/microtvm/reference-vm/arduino/base-box/base_box_provision.sh:45]: can we fix version for other boards as well?

File: apps/microtvm/reference-vm/arduino/base-box/base_box_test.sh [hunk 2]
```diff
@@ -30,11 +30,11 @@ fi
 
 board=$1
 
-pytest tests/micro/arduino/test_arduino_workflow.py --arduino-board=${board}
+pytest tests/micro/arduino/test_arduino_workflow.py --arduino-board=${board} -s
```
Comment (mehrdadh) [inline apps/microtvm/reference-vm/arduino/base-box/base_box_test.sh:33]: revert `-s`?

File: apps/microtvm/reference-vm/arduino/base-box/base_box_test.sh [hunk 3]
```diff
@@ -30,11 +30,11 @@ fi
...
 
 if [ $board == "nano33ble" ]; then
     # https://github.com/apache/tvm/issues/8730
     echo "NOTE: skipped test_arduino_rpc_server.py on $board -- known failure"
 else
-    pytest tests/micro/arduino/test_arduino_rpc_server.py --arduino-board=${board}
+    pytest tests/micro/arduino/test_arduino_rpc_server.py --arduino-board=${board} -s
```
Comment (mehrdadh) [inline apps/microtvm/reference-vm/arduino/base-box/base_box_test.sh:39]: same here.

**Review by mehrdadh — CHANGES_REQUESTED** [review]
Thanks for fixing the hack. Looks good!
It would be great if you could also fix the version for other boards. We can track the issues better this way.

File: apps/microtvm/reference-vm/arduino/base-box/base_box_test.sh (hunk 3 above)
Comment (guberti) [reply apps/microtvm/reference-vm/arduino/base-box/base_box_test.sh:39]: I'd really like to keep the `-s` flag - doing so would make it *much* easier to understand issues when the pop up on Wolfi.

File: apps/microtvm/reference-vm/arduino/base-box/base_box_provision.sh (hunk 1 above)
Comment (guberti) [reply apps/microtvm/reference-vm/arduino/base-box/base_box_provision.sh:45]: We can't cleanly fix versions for all of the others, but using a git commit hash will work. I'll make that change.

**Comment by guberti** [comment]
This PR is outdated - will close and make a new one.

**Pull request closed without merging**